controllers/clusterresources/awsendpointserviceprincipal_controller.go

```diff
@@ -175,6 +188,44 @@ func (r *AWSEndpointServicePrincipalReconciler) handleDelete(ctx context.Context
 	return nil
 }
 
+func (r *AWSEndpointServicePrincipalReconciler) startWatchStatusJob(ctx context.Context, resource *clusterresourcesv1beta1.AWSEndpointServicePrincipal) error {
+	job := r.newWatchStatusJob(ctx, resource)
+	return r.Scheduler.ScheduleJob(resource.GetJobID(scheduler.StatusChecker), scheduler.ClusterStatusInterval, job)
+}
+
+func (r *AWSEndpointServicePrincipalReconciler) newWatchStatusJob(ctx context.Context, principal *clusterresourcesv1beta1.AWSEndpointServicePrincipal) scheduler.Job {
+	return func() error {
+		_, err := r.API.GetAWSEndpointServicePrincipal(principal.Status.ID)
+		if err != nil {
+			if errors.Is(err, instaclustr.NotFound) {
+				return r.handleExternalDelete(ctx, principal)
+			}
+
+			return err
+		}
+
+		return nil
+	}
+}
```

Comment:
add a checker here for the existence of k8s resource

Reply:
Added here and for other resources